packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,665 @@
+            });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.CreateDomain: {
+        try {
+          const response = await commands.createDomain(commandOptions, {
+            domainId: req.params[0],
+            defaultRole: req.params[1],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.SetAccountQuorum: {
+        try {
+          const response = await commands.setAccountQuorum(commandOptions, {
+            accountId: req.params[0],
+            quorum: req.params[1],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.AddAssetQuantity: {
+        try {
+          const response = await commands.addAssetQuantity(commandOptions, {
+            assetId: req.params[0],
+            amount: req.params[1],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.SubtractAssetQuantity: {
+        try {
+          const response = await commands.subtractAssetQuantity(
+            commandOptions,
+            {
+              assetId: req.params[0],
+              amount: req.params[1],
+            },
+          );
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.TransferAsset: {
+        try {
+          const response = await commands.transferAsset(commandOptions, {
+            srcAccountId: req.params[0],
+            destAccountId: req.params[1],
+            assetId: req.params[2],
+            description: req.params[3],
+            amount: req.params[4],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetSignatories: {
+        try {
+          const queryRes = await queries.getSignatories(queryOptions, {
+            accountId: req.params[0],
+          });
+          return { transactionReceipt: queryRes };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetAccount: {
+        try {
+          const queryRes = await queries.getAccount(queryOptions, {
+            accountId: req.params[0],
+          });
+          return { transactionReceipt: queryRes };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaQuery.GetAccountDetail: {
+        try {
+          const queryRes = await queries.getAccountDetail(queryOptions, {
+            accountId: req.params[0],
+            key: req.params[1],
+            writer: req.params[2],
```

Comment:
Thank you for pointing out this! Eventually the request will be an interface, so the fields will automatically be optional instead of required by the array.

Reply:
Sorry for misleading you earlier. I believe that as we discussed today, we want to keep the param as of now. 
Nevertheless, the basic configuration parameters of Iroha has already been integrated as an interface.